lib/solr/request.rb

```diff
@@ -27,11 +28,8 @@ def initialize(document_type:, search_term:)
     # returns [Solr::Response]
     def run(page:, page_size:)
       solr_params = Solr::Request::EdismaxAdapter.new(self).to_h
-      solr_response = Solr.with_instrumentation('query.solr_request', solr_params: solr_params) do
-        # We should use POST method to avoid GET HTTP 413 error (request entity too large)
-        Solr::SolrCaller.call(page: page, page_size: page_size, solr_params: solr_params)
-      end
-      Solr::Response::EdismaxAdapter.new(request: self, solr_response: solr_response).to_response
+      solr_response = Solr::Query::Request.run(page: page, page_size: page_size, solr_params: solr_params)
```

Comment:
it's confusing here. You have Solr::Response and solr_response which are very different things, but look like the same. 

You should probably use raw_ to make it clear for network/unprocessed responses